source/backend/api/Services/CompensationRequisitionService.cs

```diff
@@ -107,27 +93,50 @@ public bool DeleteCompensation(long compensationId)
 
             var currentCompensation = _compensationRequisitionRepository.GetById(compensationId);
 
-           /* TODO: Fix compensation requisition
-            var currentAcquisitionStatus = GetCurrentAcquisitionStatus(currentCompensation.AcquisitionFileId);
-
-            if (!_statusSolver.CanEditOrDeleteCompensation(currentAcquisitionStatus, currentCompensation.IsDraft) && !_user.HasPermission(Permissions.SystemAdmin))
+            if (currentCompensation.AcquisitionFileId is not null)
             {
-                throw new BusinessRuleViolationException("The file you are editing is not active or hold, so you cannot save changes. Refresh your browser to see file state.");
+                var currentAcquisitionStatus = GetCurrentAcquisitionStatus((long)currentCompensation.AcquisitionFileId);
+
+                if (!_acquisitionStatusSolver.CanEditOrDeleteCompensation(currentAcquisitionStatus, currentCompensation.IsDraft) && !_user.HasPermission(Permissions.SystemAdmin))
+                {
+                    throw new BusinessRuleViolationException("The file you are editing is not active or hold, so you cannot save changes. Refresh your browser to see file state.");
+                }
             }
 
             var fileFormToDelete = _compensationRequisitionRepository.TryDelete(compensationId);
             _compensationRequisitionRepository.CommitTransaction();
+
             return fileFormToDelete;
-            */
-            return true;
         }
 
-        public IEnumerable<PimsPropertyAcquisitionFile> GetProperties(long id)
+        public IEnumerable<PimsPropertyAcquisitionFile> GetAcquisitionProperties(long id)
         {
             _logger.LogInformation("Getting properties for Compensation Requisition with id {id}", id);
             _user.ThrowIfNotAuthorized(Permissions.CompensationRequisitionView);
 
-            return _compensationRequisitionRepository.GetPropertiesByCompRequisitionId(id);
+            return _compensationRequisitionRepository.GetAcquisitionCompReqPropertiesById(id);
+        }
+
+        public IEnumerable<PimsPropertyLease> GetLeaseProperties(long id)
+        {
+            _logger.LogInformation("Getting properties for Compensation Requisition with id {id}", id);
+            _user.ThrowIfNotAuthorized(Permissions.CompensationRequisitionView);
+
+            return _compensationRequisitionRepository.GetLeaseCompReqPropertiesById(id);
+        }
+
+        public IEnumerable<PimsCompensationRequisition> GetFileCompensationRequisitions(FileTypes fileType, long fileId)
+        {
```

Comment:
permissions checks? logging?

Reply:
those are at each individual private method depending on the fileType.